### PR TITLE
Janb/compound client

### PR DESCRIPTION
**Pull request by benesjan**
# Description

Took the original implementation of compound client, removed functions which already exist in ERC4626 client, made it extend ERC4626 client and modified the remaining functions to work with wrapped cTokens instead of cTokens.

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] There are no unexpected formatting changes, or superfluous debug logs.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewers next convenience.
- [x] NatSpec documentation of all the non-test functions is present and is complete.
- [ ] Continuous integration (CI) passes.
- [ ] Command `forge coverage --match-contract MyContract` returns 100% line coverage.
- [ ] All the possible reverts are tested.

### CODE REVIEW

File: src/client/compound/compound-bridge-data.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    // To get the cToken first call cToken on the wrapper contract
+    const cTokenAddress = await this.getCToken(yieldAsset.erc20Address);
+    const cToken = ICERC20__factory.connect(cTokenAddress.toString(), this.ethersProvider);
+
+    const supplyRatePerBlock = await cToken.supplyRatePerBlock();
+    return supplyRatePerBlock.mul(blocksPerYear).toNumber() / 10 ** 16;
+  }
+
+  async getMarketSize(
```
Comment (LHerskind) [inline src/client/compound/compound-bridge-data.ts:36]: Unsure if this is the way we want to compute it. As Compound is a lending/borrowing market, if it is fully utilised, the market size from the balance of the cToken would be 0, where the more "correct" might be the total supply of the wrappers instead.

**Review by LHerskind — CHANGES_REQUESTED** [review]
Mostly looks good, don't think the way `getMarketSize` is defined is the proper way to do it though.

File: src/client/compound/compound-bridge-data.ts (hunk 1 above)
Comment (benesjan) [reply src/client/compound/compound-bridge-data.ts:36]: Good point. In AAVE I am fetching totalLiquidity which should correspond to the value we have here. I will just use the wrapper `totalSupply` value and move the function to ERC4626, ok? If you agree I would merge this and then do a separate PR with the change.

Comment (benesjan) [reply src/client/compound/compound-bridge-data.ts:36]: I could return the totalAssets value from ERC4626 but the think is that the value is wrapper specific while users are probably thinking about the protocol as a whole.
image

Comment (benesjan) [reply src/client/compound/compound-bridge-data.ts:36]: Addressed in e8c06f26b832a75a3d68a5145ba2e93d72949e60

**Review by LHerskind — APPROVED** [review]

**Pull request merged**